query-sql/src/lib.rs

```diff
@@ -290,19 +322,15 @@ impl QueryFragment for SelectQuery {
     fn push_sql(&self, out: &mut QueryBuilder) -> BuildQueryResult {
         out.push_sql("SELECT ");
         self.projection.push_sql(out)?;
-
-        out.push_sql(" FROM ");
         self.from.push_sql(out)?;
 
-        if self.constraints.is_empty() {
-            return Ok(());
+        if !self.constraints.is_empty() {
```

Comment:
If you know that `from` is `None`, you probably never need to process `constraints` or `projection`.  If you look later on, these are always trivial when `from` is `None`.  I'm suggesting you lift that "block representation" of `None, trivial, trivial` to `SelectQuery::None` or similar.